### PR TITLE
feat: add target dtype conversion

### PR DESCRIPTION
**Pull request by cristianmtr**
- add target_dtype arg
- make Jina req flexible with `~=`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
-jina[hub]==2.0.0
+jina-commons @ git+https://github.com/jina-ai/jina-commons.git#egg=jina-commons
```
Comment (slettner) [inline requirements.txt]:
```suggestion
jina-commons @ git+https://github.com/jina-ai/jina-commons.git@v0.0.1#egg=jina-commons
```

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import setuptools
-
+from pkg_resources import parse_requirements
```
Comment (slettner) [inline setup.py]: Looks unused

File: tests/integration/test_integration.py [hunk 3]
```diff
@@ -25,10 +28,13 @@ def data_generator(num_docs, numpy_image_uri):
         yield doc
 
 
-def test_use_in_flow(numpy_image_uri):
+@pytest.mark.parametrize('dtype', ['numpy.uint8', 'numpy.float32', 'numpy.float64'])
+def test_use_in_flow(numpy_image_uri, dtype):
+    os.environ['DTYPE'] = dtype
```
Comment (JoanFM) [inline tests/integration/test_integration.py]: this should be a fixture that is unset at the end

**Review by JoanFM — DISMISSED** [review]

File: tests/integration/test_integration.py (hunk 3 above)
Comment (cristianmtr) [reply tests/integration/test_integration.py]: Cab you pass a pytest.mark.parametrize parameter to a fixture?

Comment (JoanFM) [reply tests/integration/test_integration.py]: yes, with `indirect`

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,5 @@
 FROM jinaai/jina:2.0
+RUN apt-get update && apt-get install -y python3.7 python3.7-dev python3-pip git
```
Comment (numb3r3) [inline Dockerfile]:
```suggestion
FROM jinaai/jina:2.0.7-py37-standard

RUN apt-get update && apt-get install --no-install-recommends -y git \
                && rm -rf /var/lib/apt/lists/*
```

**Review by numb3r3 — DISMISSED** [review]

File: Dockerfile (hunk 4 above)
Comment (cristianmtr) [reply Dockerfile]: This needs to be done with automation, not manually. Needs to be discussed

File: tests/integration/test_integration.py (hunk 3 above)
Comment (cristianmtr) [reply tests/integration/test_integration.py]: done

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**